File: 프로그래머스/lv2/42578. 의상/의상.js
function solution(clothes) {
    var answer = 1;
    const obj = {}
    
    for(var i=0; i<clothes.length; i++){
        if(clothes[i][1] in obj){
            obj[clothes[i][1]] += 1;
        }else{
            obj[clothes[i][1]] = 2;
        }
    }
    for(var i in obj){
        answer *= obj[i];
    }

    return answer - 1;
}

File: 프로그래머스/lv2/42578. 의상/의상.py

def solution(clothes):
    answer = 1
    obj = {}
    
    for i in range(len(clothes)):
        if clothes[i][1] not in obj:
            obj[clothes[i][1]] = 2
        else:
            obj[clothes[i][1]] += 1
    
    for key in obj.keys():
        answer *= obj[key]
    
    return answer - 1